src/core/server/plugins/plugin_context.ts

```diff
@@ -107,8 +107,8 @@ export function createPluginInitializerContext(
        * @param ConfigClass A class (not an instance of a class) that contains a
        * static `schema` that we validate the config at the given `path` against.
        */
-      create() {
-        return coreContext.configService.atPath(pluginManifest.configPath);
+      create<T>() {
+        return coreContext.configService.atPath<T>(pluginManifest.configPath).pipe(shareReplay(1));
```

Comment:
The spaces plugin takes a config value for every `getScopedClient` call (every incoming request). This does config validation for this config path each time. https://github.com/elastic/kibana/blob/c5e446adbdde5c23bc06f326792aaff3e35dd937/x-pack/plugins/spaces/server/spaces_service/spaces_service.ts#L75-L82

We could optimize the plugin to not take a new config value every time but only if a new value was emitted, but this felt like a better fix that will probably benefit other places.